llvm/lib/Target/AMDGPU/SIPreEmitPeephole.cpp

```diff
@@ -304,11 +311,23 @@ bool SIPreEmitPeephole::getBlockDestinations(
   return true;
 }
 
-bool SIPreEmitPeephole::mustRetainExeczBranch(
-    const MachineBasicBlock &From, const MachineBasicBlock &To) const {
-  unsigned NumInstr = 0;
-  const MachineFunction *MF = From.getParent();
+bool SIPreEmitPeephole::mustRetainExeczBranch(const MachineBasicBlock &Head,
+                                              const MachineBasicBlock &From,
+                                              const MachineBasicBlock &To) {
 
+  assert(is_contained(Head.successors(), &From));
+
+  const auto *FromIt = find(Head.successors(), &From);
+  assert(FromIt != Head.succ_end());
+
+  auto BranchProb = Head.getSuccProbability(FromIt);
+  assert(!BranchProb.isUnknown());
+  uint64_t BranchTakenCost = TII->getSchedModel().computeInstrLatency(
+      &*Head.getFirstTerminator(), false);
+  constexpr uint64_t BranchNotTakenCost = 1;
```

Comment:
Unfortunately the cycle counts aren't normalized across targets. Pre-gfx10 the cycle counts are in units of multiples of 4 (we should probably fix this)

Reply:
I think that keeping 1 may be safe in this case, since it only makes the transformation more unlikely.

Reply:
Of course it's safe to pick any number, but more precision may yield better results